x/participationrewards/keeper/distribution.go

```diff
@@ -40,9 +40,12 @@ func (k Keeper) getRewardsAllocations(ctx sdk.Context) rewardsAllocation {
 		k.Logger(ctx).Info("nothing to distribute...")
 
 		// create snapshot of current intents for next epoch boundary
+		// requires intents to be set, no intents no snapshot...
+		// we need to ensure the snapshot is only taken once;
 		// TODO: this needs to be verified as it currently does not trigger anymore
 		for _, zone := range k.icsKeeper.AllZones(ctx) {
 			for _, di := range k.icsKeeper.AllOrdinalizedIntents(ctx, zone, false) {
+				k.Logger(ctx).Info("TRIGGERED")
```

Comment:
left over debug logging?

Reply:
we have to ensure that intent snapshot is only taken once and not twice...

Reply:
this log will trigger once for every user, for ever zone - I suspect it'll be too noisy to be useful?

Reply:
Ok, it seems this is the bootstrapping of intent snapshots...
When we have no intents set the snapshot will not trigger, however if we have intents and a zero module balance we want to take a snapshot of intents to utilize at the next epoch boundary.

So this will not trigger again unless the module balance remains zero (under what reasonable circumstances could this be possible?)